### PR TITLE
fix: [ANDROAPP-3475] Sending jira issue from app is not working

### PR DESCRIPTION
**Pull request by Balcan**
## Description
**Note**: Candidate for 2.3.1
[ jira issue ](https://jira.dhis2.org/browse/ANDROAPP-3475)

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: app/src/main/java/org/dhis2/data/jira/JiraIssueListResponse.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    val issues: List<JiraIssue>
+) {
+    companion object {
+        fun fromJson(response: String?): JiraIssueListResponse {
+            return try {
+                Gson()
+                    .fromJson(
+                        response,
+                        JiraIssueListResponse::class.java
+                    )
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/data/jira/JiraIssueListResponse.kt:18]: Do you think is a good idea to add a ConverterFactory(GsonConverterFactory.create()) to retrofit initialization and avoid this?

File: app/src/main/java/org/dhis2/usescases/jira/JiraRepository.kt [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                        onResponse(JiraIssuesResult(issueList))
+                    } else {
+                        onResponse(JiraIssuesResult(errorMessage = response.message()))
+                    }
+                }
+
+                override fun onFailure(call: Call<ResponseBody>, t: Throwable) {
+                    onResponse(JiraIssuesResult(errorMessage = t.localizedMessage))
+                }
+            }
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/usescases/jira/JiraRepository.kt:62]: What do you think about removing the enqueue callback and make it sync, then using IO RxJava? You could remove the callback. 

Thoughts?

File: app/src/main/java/org/dhis2/usescases/jira/JiraViewModel.kt [hunk 3]
```diff
@@ -1,208 +1,138 @@
...
+        } else {
+            issueMessage.value = resources.jiraIssueSentErrorMessage()
+        }
     }
 
-    override fun onJiraPassChanged(s: CharSequence, start: Int, before: Int, count: Int) {
-        pass.value = s.toString()
+    @VisibleForTesting
+    fun handleThrowableResponse(throwable: Throwable) {
+        issueMessage.value = throwable.localizedMessage
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/usescases/jira/JiraViewModel.kt:117]: I don't think you need to use VisibleforTesting. With Mockito you can mocks callbacks and choose either success or error, then you can make the assertions. Let me know if you need help with this !

**Comment by JaimeToca** [comment]
@Balcan Great Job, really nice PR

File: app/src/main/java/org/dhis2/data/jira/JiraIssueListResponse.kt (hunk 1 above)
Comment (Balcan) [reply app/src/main/java/org/dhis2/data/jira/JiraIssueListResponse.kt:18]: yes, that would be nice 👍

File: app/src/main/java/org/dhis2/usescases/jira/JiraRepository.kt (hunk 2 above)
Comment (Balcan) [reply app/src/main/java/org/dhis2/usescases/jira/JiraRepository.kt:62]: I'm going to look at this. My only concern is dealing with the disposables and the viewmodel lifecycle at the same time...

File: app/src/main/java/org/dhis2/usescases/jira/JiraViewModel.kt (hunk 3 above)
Comment (Balcan) [reply app/src/main/java/org/dhis2/usescases/jira/JiraViewModel.kt:117]: true

**Review by JaimeToca — APPROVED** [review]

**Pull request merged**